zerver/views/auth.py

```diff
@@ -123,6 +127,7 @@ def confirm_email_change(request: HttpRequest, confirmation_key: str) -> HttpRes
         'new_email': new_email,
         'old_email': old_email,
     }
+    do_login(request, user_profile)
```

Comment:
This is to improve the user experience, so they don't have auth again right after they just authed, right? I think that makes sense, so we might want to give it more extra thought about how safe this will be, inside a function like this. Can you move this change to its own commit, as the last commit?

Reply:
Yes, (though it is unnecessary if the user is already logged in.) I do not see any security issue as we call this function at two places where user is confirmed to have been authed.
* We can be sure with social auth and remote sso flows because this is being called in `login_or_register_remote_user` which is only called after successful auth afaik.
* I'm not 100% sure about the email auth case as you pointed out that not having `SimpleTemplateResponse` isn't a better check to assert that the user is authenticated which I'm convinced that it could be unsafe.

If we're convinced to change a user's email(which I think, is a stronger action than logging in) we can log them in at the same time i.e just after changing email.
Made it a separate commit.

Reply:
For email+password auth case, now I've changed it to check `request.user.is_authenticated` before proceeding to `confirm_email_change` so I think we are on a safe side to log user in or take to `confirm_email_change`.